apps/root/src/pages/earn/components/all-strategies-table/components/filters/index.tsx

```diff
@@ -0,0 +1,325 @@
+    };
+
+    const farmsFilter: FilterControl<string> = {
+      options: strategiesParameters.farms.map((farm) => ({
+        value: farm.id,
+        label: <Typography variant="bodySmallSemibold">{farm.name}</Typography>,
+        searchParams: [farm.name],
+      })),
+      filteredOptions: strategiesFilters.farms,
+      summaryLabel: intl.formatMessage(
+        defineMessage({
+          defaultMessage: 'Protocol',
+          description: 'all-strategies-table.filters.protocol',
+        })
+      ),
+      handleFilterChange: (filter) => dispatch(setFarmFilter(filter)),
+    };
+
+    const networksFilter: FilterControl<number> = {
+      options: strategiesParameters.networks.map((network) => ({
+        value: network.chainId,
+        label: <Typography variant="bodySmallSemibold">{network.name}</Typography>,
+        searchParams: [network.name, network.chainId.toString()],
+      })),
+      filteredOptions: strategiesFilters.networks,
+      summaryLabel: intl.formatMessage(
+        defineMessage({
+          defaultMessage: 'Network',
+          description: 'all-strategies-table.filters.network',
+        })
+      ),
+      handleFilterChange: (filter) => dispatch(setNetworkFilter(filter)),
+    };
+
+    const yieldTypesFilter: FilterControl<StrategyYieldType> = {
+      options: strategiesParameters.yieldTypes.map((yieldType) => ({
+        value: yieldType.value,
+        label: <Typography variant="bodySmallSemibold">{yieldType.label}</Typography>,
+        searchParams: [yieldType.label],
+      })),
+      filteredOptions: strategiesFilters.yieldTypes,
+      summaryLabel: intl.formatMessage(
+        defineMessage({
+          defaultMessage: 'Yield Type',
+          description: 'all-strategies-table.filters.yield-type',
+        })
+      ),
+      handleFilterChange: (filter) => dispatch(setYieldTypeFilter(filter)),
+      hideSearch: true,
+    };
+
+    const guardiansFilter: FilterControl<string> = {
+      options: strategiesParameters.guardians.map((guardian) => ({
+        value: guardian.id,
+        label: <Typography variant="bodySmallSemibold">{guardian.name}</Typography>,
+        searchParams: [guardian.name],
+      })),
+      filteredOptions: strategiesFilters.guardians,
+      summaryLabel: intl.formatMessage(
+        defineMessage({
+          defaultMessage: 'Guardians',
+          description: 'all-strategies-table.filters.guardians',
+        })
+      ),
+      handleFilterChange: (filter) => dispatch(setGuardianFilter(filter)),
+    };
+
+    return [networksFilter, assetFilter, rewardsFilter, farmsFilter, yieldTypesFilter, guardiansFilter];
+  }, [intl, strategiesFilters, strategiesParameters]);
+
+  return (
+    <>
+      <Button onClick={handleOpen} disabled={isLoading} variant="outlined" endIcon={<KeyboardArrowDownIcon />}>
+        <FormattedMessage defaultMessage="Filters" description="all-strategies-table.filters" />
```

Comment:
can you check with emi if this works for him?

Reply:
Sure, we agreed to use a hierarchical structure from now on. In this case: `earn.all-strategies-table.filters`

Reply:
awesome!